src/test/ui/or-patterns/bindings-runpass-2.rs

```diff
@@ -0,0 +1,32 @@
+// run-pass
+
+#![feature(or_patterns)]
+
+fn or_at(x: Result<u32, u32>) -> u32 {
+    match x {
+        Ok(x @ 4) | Err(x @ (6 | 8)) => x,
```

Comment:
In an ideal world, this test would be more self-documenting (i.e. it would somehow embed in its values what the cases are that are being tested), in the hopes that a reader would, on cursory overview, conclude "yes this is obviously exhaustive and correct."

But I can understand the time investment to write such a test is non-trivial. So lets settle for this for now.